htgotts.go

```diff
@@ -68,46 +73,51 @@ func (speech *Speech) Speak(text string) error {
 	return speech.PlaySpeechFile(fileName)
 }
 
-/**
- * Create the folder if does not exists.
- */
+// Read downloads speech and plays it without writing to disk
+func (speech *Speech) Read(text string) error {
+	buf, err := speech.fetch(text)
+	if err != nil {
+		return err
+	}
+	return speech.Handler.PlayBuf(buf)
+}
+
+// Create the folder if does not exists.
 func (speech *Speech) createFolderIfNotExists(folder string) error {
 	dir, err := os.Open(folder)
 	if os.IsNotExist(err) {
-		return os.MkdirAll(folder, 0700)
+		return os.MkdirAll(folder, 0o700)
```

Comment:
Please, remove this "o" from the permission code.

Reply:
In the first line, 0700, the prefix "0" indicates an octal literal, and the digits "700" represent the permission value in octal. This octal value translates to rwx------ in the symbolic representation, which means the owner has read, write, and execute permissions, while others have no permissions.

In the second line, 0o700, the prefix "0o" is an alternative representation introduced in Go 1.13 to make it more explicit that the value is an octal literal. The digits "700" still represent the permission value in octal, and it has the same meaning as in the first line.

I think the change was made by my language server though, not me, so it's probably worth defining project specific settings (assuming you can do so for gopls) if this refactor is to be avoided. I for one think it helps though.